include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1049 @@
+    template <class A = std::allocator<char>>
+    BOOST_URL_CXX20_CONSTEXPR
+    std::basic_string<char, std::char_traits<char>, A>
+    to_string() const
+    {
+        std::basic_string<
+            char, std::char_traits<char>, A> str(size(), ' ');
+        return to_string(str);
+    }
+
+    /** Returns a basic_string
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class A>
+    BOOST_URL_CXX20_CONSTEXPR explicit
+    operator std::basic_string<char, std::char_traits<char>, A>() const
+    {
+        return to_string<A>();
+    }
+
+    /** Appends the decoded characters to a container
+
+        This overload allows the caller to recycle
+        the input string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class T>
+    BOOST_URL_CXX20_CONSTEXPR
+    T&
+    append_to(T& str) const;
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if two strings are equal after percent decoding
+      */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /// @copydoc operator==
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /// @copydoc operator==
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) == 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if two strings are not equal after percent decoding
+      */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) != 0;
+    }
+
+    /// @copydoc operator!=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) != 0;
+    }
+
+    /// @copydoc operator!=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) != 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) < 0;
+    }
+
+    /// @copydoc operator<
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<(
```

Comment:
What happens if you write
```
decoded_view dv;
if( dv < "sixteen" )
```
? I haven't looked at the tests, does this compile?

Reply:
Yes. I included it in the tests.